src/metkit/api/metkit_c.cc

```diff
@@ -0,0 +1,348 @@
+    metkit_paramiterator_t(std::vector<std::string> vec) :
+        vector_(std::move(vec)), iterator_(vector_.begin()) {}
+
+    int next() {
+        if (iterator_ == vector_.end()) {
+            return METKIT_ITERATION_COMPLETE;
+        }
+        ++iterator_;
+
+        return iterator_ == vector_.end() ? METKIT_ITERATION_COMPLETE : METKIT_SUCCESS;
+    }
+
+    void current(const char** param) {
+        ASSERT(iterator_ != vector_.end());
+        *param = iterator_->c_str();
+    }
+
+private: 
+    std::vector<std::string> vector_;
+    std::vector<std::string>::iterator iterator_;
+};
+
+// ---------------------------------------------------------------------------------------------------------------------
+//                           ERROR HANDLING
+
+}  // extern "C"
+
+static thread_local std::string g_current_error_string;
+
+const char* metkit_get_error_string(int) {
+    return g_current_error_string.c_str();
+}
+
+int innerWrapFn(std::function<int()> f) {
+    return f();
+}
+
+int innerWrapFn(std::function<void()> f) {
+    f();
+    return METKIT_SUCCESS;
+}
+
+template <typename FN>
+[[nodiscard]] int tryCatch(FN&& fn) {
+    try {
+        return innerWrapFn(fn);
+    }
+    catch (const eckit::UserError& e) {
+        eckit::Log::error() << "User Error: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR_USER;
+    }
+    catch (const eckit::AssertionFailed& e) {
+        eckit::Log::error() << "Assertion Failed: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR_ASSERT;
+    }
+    catch (const eckit::Exception& e) {
+        eckit::Log::error() << "METKIT Error: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR;
+    }
+    catch (const std::exception& e) {
+        eckit::Log::error() << "Unknown Error: " << e.what() << std::endl;
+        g_current_error_string = e.what();
+        return METKIT_ERROR_UNKNOWN;
+    }
+    catch (...) {
+        eckit::Log::error() << "Unknown Error!" << std::endl;
+        g_current_error_string = "<unknown>";
+        return METKIT_ERROR_UNKNOWN;
+    }
+}
+
+extern "C" {
+
+// -----------------------------------------------------------------------------
+//                           HELPERS
+// -----------------------------------------------------------------------------
+
+int metkit_version(const char** version) {
+    *version = metkit_version_str();
+    return METKIT_SUCCESS;
+}
+
+int metkit_vcs_version(const char** sha1) {
+    *sha1 = metkit_git_sha1();
+    return METKIT_SUCCESS;
+}
+
+int metkit_initialise() {
+    return tryCatch([] {
+        static bool initialised = false;
+
+        if (initialised) {
+            eckit::Log::warning()
+                << "Initialising Metkit library twice" << std::endl;
+        }
+
+        if (!initialised) {
+            const char* argv[2] = {"metkit-api", 0};
```

Comment:
nullptr instead of 0. Why is this `const` if a line below the const-cast takes place?